packages/core/src/repositories/projectsRepository.ts

```diff
@@ -48,4 +56,57 @@ export class ProjectsRepository extends Repository<typeof tt, Project> {
 
     return Result.ok(result)
   }
+
+  async findAllActiveDocumentsWithAgreggatedData() {
+    const lastMergedCommit = this.db.$with('lastMergedCommit').as(
+      this.db
+        .select({
+          projectId: commits.projectId,
+          maxVersion: max(commits.version).as('maxVersion'),
+        })
+        .from(commits)
+        .where(isNotNull(commits.mergedAt))
+        .groupBy(commits.projectId),
+    )
+    const aggredatedData = this.db.$with('aggredatedData').as(
+      this.db
+        .with(lastMergedCommit)
+        .select({
+          id: this.scope.id,
+          documentCount: count(documentVersions.id).as('documentCount'),
+          lastCreatedAtDocument: max(documentVersions.createdAt).as(
+            'lastCreatedAtDocument',
+          ),
+        })
+        .from(this.scope)
+        .innerJoin(commits, eq(commits.projectId, this.scope.id))
+        .innerJoin(
+          lastMergedCommit,
+          and(
+            eq(lastMergedCommit.projectId, this.scope.id),
+            eq(commits.version, lastMergedCommit.maxVersion),
+          ),
+        )
+        .innerJoin(documentVersions, eq(documentVersions.commitId, commits.id))
+        .where(isNull(this.scope.deletedAt))
+        .groupBy(this.scope.id),
+    )
+
+    const result = await this.db
+      .with(aggredatedData)
+      .select({
+        id: this.scope.id,
+        name: this.scope.name,
+        createdAt: this.scope.createdAt,
+        updatedAt: this.scope.updatedAt,
+        workspaceId: this.scope.workspaceId,
+        deletedAt: this.scope.deletedAt,
+        documentCount: aggredatedData.documentCount,
+        lastCreatedAtDocument: aggredatedData.lastCreatedAtDocument,
```

Comment:
Here is a dilema. Before it was 2 calls. 1 in server for projects and othe in frontend for documents and do the match. All is fine with that approach except the hidratation issue with the dates.

But what do you think about doing the thing ssr?

Reply:
repository shouldn't be conditioned by frontend serialization concerns tbh, that's the function of presenters.

Reply:
I'll do an spread here. In the end we were already sending all from projects